File: frontend/src/api/commandCenter.ts
export interface CommandCenterStep {
  category_id: string;
  category_label: string;
  status: "consulted" | "locked";
  summary: string;
}

export interface AskCommandCenterResponse {
  session_id: string;
  steps: CommandCenterStep[];
  answer: string;
}

export interface CommandCenterSessionSummary {
  id: string;
  title: string;
  updated_at: string;
}

export interface CommandCenterMessage {
  role: "staff" | "agent";
  content: string;
  steps: CommandCenterStep[];
  created_at: string;
}

export interface CommandCenterSessionDetail {
  id: string;
  title: string;
  messages: CommandCenterMessage[];
}

type AuthedFetch = <T>(path: string, init?: RequestInit) => Promise<T>;

// Matches backend/src/router/command_center/command_center_router.py.
export function askCommandCenter(authedFetch: AuthedFetch, question: string, sessionId?: string | null) {
  return authedFetch<AskCommandCenterResponse>("/api/v1/command-center/ask", {
    method: "POST",
    body: JSON.stringify({ question, session_id: sessionId ?? null })
  });
}

export function listCommandCenterSessions(authedFetch: AuthedFetch) {
  return authedFetch<CommandCenterSessionSummary[]>("/api/v1/command-center/sessions");
}

export function getCommandCenterSession(authedFetch: AuthedFetch, sessionId: string) {
  return authedFetch<CommandCenterSessionDetail>(`/api/v1/command-center/sessions/${sessionId}`);
}

File: backend/src/schemas/command_center.py
from __future__ import annotations
from pydantic import BaseModel
from typing import Literal
from uuid import UUID
from datetime import datetime


class AskCommandCenterRequest(BaseModel):
    question: str
    # Omit to start a new session; pass an existing one to continue that
    # conversation's history (see CommandCenterService.ask).
    session_id: UUID | None = None


class CommandCenterStep(BaseModel):
    category_id: str
    category_label: str
    status: Literal["consulted", "locked"]
    summary: str


class AskCommandCenterResponse(BaseModel):
    session_id: UUID
    steps: list[CommandCenterStep]
    answer: str


class CommandCenterMessage(BaseModel):
    role: Literal["staff", "agent"]
    content: str
    steps: list[CommandCenterStep] = []
    created_at: datetime


class CommandCenterSessionSummary(BaseModel):
    id: UUID
    title: str
    updated_at: datetime


class CommandCenterSessionDetail(BaseModel):
    id: UUID
    title: str
    messages: list[CommandCenterMessage]
